modules/social_features/social_comment/src/Entity/Access/CommentQueryAccessHandler.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace Drupal\social_comment\Entity\Access;
+
+use Drupal\Core\Entity\EntityPublishedInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\entity\QueryAccess\ConditionGroup;
+use Drupal\entity\QueryAccess\QueryAccessHandlerBase;
+use Drupal\user\EntityOwnerInterface;
+
+/**
+ * Controls query access for comment entities.
+ *
+ * @see \Drupal\entity\QueryAccess\QueryAccessHandler
+ */
+class CommentQueryAccessHandler extends QueryAccessHandlerBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildConditions($operation, AccountInterface $account) {
```

Comment:
We can't because in the `QueryAccessHandlerBase` used `uid` entity key, but we don't have this key for the comment entity, we have `owner` instead, so this is the reason.

Reply:
Ah I see, that's a bug that's fixed in https://www.drupal.org/project/entity/issues/3065178 Fixed as of 1.0 🤔 

The update from 1.0-rc2 -> 1.0-rc3 was done in #1414 It appears that Open Social itself is holding back the version of the Entity module. Looking at [the release notes](https://www.drupal.org/project/entity/releases/8.x-1.0) I think we can at least update to 1.0 if @ronaldtebrake agrees.

Interestingly the [1.1 release of Entity API](https://www.drupal.org/project/entity/releases/8.x-1.1) contains exactly [the change proposed in this PR](https://www.drupal.org/node/3086409). So I think updating to 1.1 would be even preferable over merging this PR. The full change can be seen in https://git.drupalcode.org/project/entity/commit/eff479b

@ronaldtebrake Do you know if our upgrade for group is blocked on work caused by Entity API 1.1? If not then perhaps we can already update to the latest Entity API version now and use that to make the group update slightly easier?

Reply:
The latest version of Group actually requires 1.2 of entity api :) so we need that update once we want to do the Group update to the latest version, so I'm all for it.

The only reason why we didnt do it back then was capacity, no restrictions that I know of. 
Also bear in mind the test coverage isn't perfect so coming from RC going to 1.2 needs some proper reviews and ideally a separate PR.

Reply:
So, the distro team will do this update, or I should? 
Because it sounds like we have to do proper testing for the groups and probably other entities after an update.

Reply:
Btw, I think we can go with these changes, and this PR not blocked by updating, and after update `entity` and `group` modules to the latest version we can just return the parent in mentioned methods.
I say this because I'm afraid that update can take some time and all related PR's will be blocked.

Reply:
> Btw, I think we can go with these changes, and this PR not blocked by updating, and after update entity and group modules to the latest versions we can just return the parent in mentioned methods.
I say this because I'm afraid that update can take some time and all related PR's will be blocked.

I agree. However, there's also still quite a bit of work needed in this PR and for our own code we'd need test coverage (rather than being able to rely on test coverage from the other module) to ensure we don't introduce security issues here. So I'm not sure doing the update is necessarily more work (and it would be a good benefit to Open Social as a whole).